sycl/doc/SYCLPluginInterface.md

```diff
@@ -41,12 +40,45 @@ once before any actual offload is attempted.
 
 ### Plugin discovery
 
-Plugins are physically dynamic libraries stored somewhere in the system where
-the SYCL runtime runs. TBD - design and describe the process in details.
+Plugins are physically dynamic libraries or shared objects.
+The process to discover plugins will follow the following guidelines.
+
+The SYCL Runtime will search for plugins at env LD_LIBRARY_PATH location on
+Linux and env LIB on Windows, with names in the format of libpiXXX.so and
+query them. An extension to the search mechanism is to use a configuration file
+which lists all the available plugins and their locations. 
+This file can be stored along with SYCL Runtime.
+Plugins should expose the information of supported PI API version as a constant
+string, so it can be read without loading the library.
+The user can select/disable a specifc plugin with an environment variable.
+The Plugin Interface queries the plugins on the supported PI version and check
+for compatibility.
+The Plugin Interface then queries each plugin for all Interface Functions
+corresponding function pointers and populate a list of these pointers for
+each plugin. As an extension, the plugin can provide a populated function
+pointer table/struct for each device attached to the plugin.
+A trace mechanism should be provided to log the discovery/connection/device
+enumeration process. Eg: Display all the plugins being discovered, their
+information and supported PI version. List attached devices and their properties.
+
+ TBD - design and describe the process in detail.
 
 #### Plugin binary interface
-TBD - list and describe all the symbols plugin must export in order to be picked
-up by the SYCL runtime for offload.
+Currently the plugins should export all the symbols that are present in the file
+pi.h. In the future, this document will list the minimum set of Interface APIs
+to be supported by Plugins. This will also require adding functionality to SYCL
+Runtime to work with such limited functionality plugins.
+
+TBD - list and describe all the symbols that a plugin must export in order to
```

Comment:
Probably here you can put a link to the corresponding section of the document which describes PI.